common/src/main/java/com/revenuecat/purchases/common/CustomerInfoFactories.kt

```diff
@@ -65,3 +65,42 @@ fun JSONObject.buildCustomerInfo(): CustomerInfo {
         originalPurchaseDate = originalPurchaseDate
     )
 }
+
+/**
+ * Parses expiration dates in a JSONObject
+ * @throws [JSONException] If the json is invalid.
+ */
+private fun JSONObject.parseExpirations(): Map<String, Date?> {
+    return parseDates("expires_date")
+}
+
+/**
+ * Parses purchase dates in a JSONObject
+ * @throws [JSONException] If the json is invalid.
+ */
+private fun JSONObject.parsePurchaseDates(): Map<String, Date?> {
+    return parseDates("purchase_date")
+}
+
+/**
+ * Parses dates that match a JSON key in a JSONObject
+ * @param jsonKey Key of the dates to deserialize from the JSONObject
+ * @throws [JSONException] If the json is invalid.
+ */
+private fun JSONObject.parseDates(jsonKey: String): HashMap<String, Date?> {
+    val expirationDates = HashMap<String, Date?>()
+
+    val it = keys()
+    while (it.hasNext()) {
+        val productId = it.next()
+        val basePlanId = getJSONObject(productId).optString("product_plan_identifier")
+            .takeIf { it.isNotEmpty() }?.split(":")?.firstOrNull()
```

Comment:
Why the split by ":" here? The plan identifier should just be basePlanId without anything else, right?
https://github.com/RevenueCat/khepri/blob/d1aec0d684e21a682b55234447711fcd4d8c5f82/khepri/serializers/subscriber_serializer.py#L243

Reply:
Oh, is it?! I assumed the offer also was present in it but I'm not sure why 🤦‍♂️

I will remove that then!